[박소현] Week5

ParkSohyunee opened the pull request:
## 요구사항

### 기본

- [x] 이메일 input에서 focus out 할 때, 값이 없을 경우 input에 빨강색 테두리와 아래에 “이메일을 입력해주세요.” 빨강색 에러 메세지가 보이나요?
- [x] 이메일 input에서 focus out 할 때, 이메일 형식에 맞지 않는 값이 있는 경우 input에 빨강색 테두리와 아래에 “올바른 이메일 주소가 아닙니다.” 빨강색 에러 메세지가 보이나요?
- [x] 이메일 input에서 focus out 일 때, input 값이 test@codeit.com 일 경우 input에 빨강색 테두리와 아래에 “이미 사용 중인 이메일입니다.” 빨강색 에러 메세지가 보이나요?
- [x] 비밀번호 input에서 focus out 할 때, 값이 8자 미만으로 있거나 문자열만 있거나 숫자만 있는 경우, input에 빨강색 테두리와 아래에 “비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.” 빨강색 에러 메세지가 보이나요?
- [x] 비밀번호 input과 비밀번호 확인 input의 값이 다른 경우, 비밀번호 확인 input에 빨강색 테두리와 아래에 “비밀번호가 일치하지 않아요.” 빨강색 에러 메세지가 보이나요?
- [x] 회원가입을 실행할 경우, 문제가 있는 경우 문제가 있는 input에 빨강색 테두리와 에러 메세지가 보이나요?
- [x] 이외의 유효한 회원가입 시도의 경우, “/folder”로 이동하나요?
- [x] 회원가입 버튼 클릭 또는 Enter키 입력으로 회원가입 되나요?
- [x] 이메일, 비밀번호, 비밀번호 확인 input에 에러 관련 디자인을 Components 영역의 에러 케이스로 적용했나요?

### 심화

- [x] 눈 모양 아이콘 클릭시 비밀번호의 문자열이 보이기도 하고, 가려지기도 하나요?
- [x] 비밀번호의 문자열이 가려질 때는 눈 모양 아이콘에는 사선이 그어져있고, 비밀번호의 문자열이 보일 때는 사선이 없는 눈 모양 아이콘이 보이나요?
- [x] 로그인, 회원가입 페이지에 공통적으로 사용하는 로직이 있다면, 반복하지 않고 공통된 로직을 모듈로 분리해 사용했나요?

## 주요 변경사항

- 회원가입 페이지 유효성 검사 구현
- 이메일, 비밀번호 관련 유효성 검사 로직, 비밀번호 가리기 기능 공통 모듈로 분리

## 스크린샷

![스크린샷 2023-10-07 오전 10 54 21](https://github.com/codeit-bootcamp-frontend/1-Weekly-Mission/assets/124856726/3c685990-f13d-4d29-9c73-a80bd1950d7e)

## 멘토에게

- 전체적으로 피드백 부탁드립니다. 감사합니다.
- 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

ParkSohyunee committed:
style: 프리티어 재설정에 따른 줄바꿈 변경

ParkSohyunee committed:
feat: signup - 이메일, 비밀번호, 비밀번호 확인 유효성 검사 기능 추가

ParkSohyunee committed:
feat: signup - 회원가입 제출 시 유효성 검사 기능 추가

ParkSohyunee committed:
refactor: signin - 공통으로 사용하는 로직 리팩토링

llucas93 commented on Linkbrary/src/signin.js:
```diff
@@ -12,52 +14,54 @@ const passwordError = document.getElementById("passwordError");
+  if (!value) {
+    emailError.textContent = "이메일을 입력해주세요";
+    addErrorClass(email);
+    return;
```
(의견) 사소하지만 함수의 반환형이 bool 형이니 아닐땐 false 로 통일해주는건 어떨까요?

ParkSohyunee replied:
안녕하세요 멘토님, 상세 피드백 주셔서 감사드립니다.
지난번 멘토링 시간 의견 및 전달 주신 패드백 참고하여 리팩토링 완료 하였습니다.
주요 변경사항은 다음과 같습니다.

- 모듈화 (공통으로 사용하는 태그 및 함수)
- 이메일, 비밀번호 유효성 검사 로직 개선 및 return 시 bool 타입 명시
- 로그인 또는 회원가입 버튼 클릭 시 로직 간소화

향후 더 좋은 방향으로 개선할 수 있는지 고민해보겠습니다! 😆

llucas93 commented:
고생하셨습니다 소현님 👍 

signin, signup의 형태가 거의 비슷하다보니 코드중복이 많은데요
이부분 관련해서 멘토링 시간때 자세히 토의해보면 좋을것같습니다

ParkSohyunee committed:
refactor: 피드백 반영하여 리팩토링
- 공통으로 사용하는 태그 분리
- utils 폴더에 공통 기능 분리
- 이메일, 비밀번호 유효성 검사 로직 개선
- return 시 bool 타입 명시
